src/lib.rs

```diff
@@ -37,11 +37,26 @@ mod errors {
                 description("unrecognizable charset")
                 display("uchardet was unable to recognize a charset")
             }
-            DataHandlingError {
-                description("data handling error")
-                display("uchardet failed to handle the supplied data (out of memory?)")
+            OutOfMemory {
+                description("out of memory error")
+                display("uchardet ran out of memory")
+            }
+            Other(int: i32) {
+                description("unknown error")
+                display("uchardet returned unknown error {}", int)
+            }
+        }
+    }
+
+    impl ErrorKind {
+        pub fn from_nsresult(nsresult: ::ffi::nsresult) -> ErrorKind {
```

Comment:
(This comment is just a remark about something I noticed, not a feature request!)

You know, I think it might actually be possible to fix error-chain to _support_ doc comments. Comments starting with `///` actually get transformed to `#[doc...]` attributes before macros see them (if I recall correctly), so the macro could move them around and attach them to the right `enum` members on output.